libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerOptions.cs

```diff
@@ -111,5 +111,15 @@ public QnAMakerOptions()
         /// <seealso cref="RankerTypes"/>
         [JsonProperty("rankerType")]
         public string RankerType { get; set; }
+
+        /// <summary>
+        /// Gets or Sets a value of Compound Filter for Metadata Join Expression.
+        /// </summary>
+        /// [JsonProperty("strictFiltersCompoundOperationType")]
+        /// <value>
+        /// A value used for join operation of Metadata <see cref="Metadata"/>.
+        /// </value>
+        [JsonProperty("strictFiltersCompoundOperationType")]
+        public StrictFiltersCompoundOperationType StrictFiltersCompoundOperationType { get; set; }
```

Comment:
>StrictFiltersCompoundOperationType [](start = 50, length = 34)

This is a pretty heinous name.  It does correspond to the REST API, but from an SDK standpoint something like "MetadataJoinOperator" would be clearer.

Reply:
@chrimc62 
Thanks you chris for name suggestion.

we already have another field named 'StrictFilters'  in this class. So, as this tied to it, better to keep same prefix 'StrictFilters'

'CompoundOperationType' word, we borrowed from Azure Search, as its actually API/SDK, as QnA Maker exposed same azure search filter to customer
https://docs.microsoft.com/en-us/azure/search/search-filters

Reply:
I would prefer StrictFiltersJoinOperator and the type JoinOperator.  This is much less wordy and compound is not adding to understanding.

---
In reply to: [486672715](https://github.com/microsoft/botbuilder-dotnet/pull/4446#discussion_r486672715) [](ancestors = 486672715)